compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -146,13 +146,13 @@ namespace std
         #{initMetadata}
     };
 
-    #{template}inline void swap(#{structName}& left, #{structName}& right)
+    #{template}inline void swap(#{qualifiedStructName}& left, #{qualifiedStructName}& right)
     {
         left.swap(right);
     }|]
       where
         template = CPP.template s
-        structName = CPP.structName s
+        qualifiedStructName = CPP.qualifiedStructName cpp s
```

Comment:
I originally tried to make this change the same way I did d1efb601, but the generated code for the Box<T> in `cpp/inc/bond/core/bond.bond` didn't compile. It appears that `Box::swap(Box left, ...)` is okay, but `Box::swap(::bond::Box left ...)` fails to build because `::bond::Box` is generic. Looks like we don't generate apply functions for parameterized types, so it's not strictly necessary that we carry this change over there.

Additionally, for reasons that are not immediately obvious to me, Types deals in strings, but Apply deals in Text. To use this in Apply, I'd either have to make this function break with its neighbors and return Text, or I'd have to reTextify the string that I just unTextified. I'd like to leave it as is.